src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsVuapOasis.java

```diff
@@ -276,4 +279,40 @@ public void vuapClickToPlayShouldStartPlayingAdvertisementAfterClickOnPlayIcon(P
 
     VuapAssertions.verifyVideoPlay(vuap);
   }
+
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsVuapClickToPlayDesktop"}
+  )
+  public void ABCDShouldRequestForMEGAAdUnitVAST() throws UnsupportedEncodingException {
+    final String adUnit = "/5441/wka1a.VIDEO/abcd/desktop/oasis-article/_project43-life";
+    checkRequestForAdUnit(AdsDataProvider.UAP_ABCD_PAGE, adUnit);
+  }
+
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsVuapClickToPlayDesktop"}
+  )
+  public void HiViBTFShouldRequestForMEGAAdUnitVAST() throws UnsupportedEncodingException {
+    final String adUnit = "/5441/wka1a.VIDEO/uap_bfaa/desktop/oasis-article/_project43-life";
+    checkRequestForAdUnit(AdsDataProvider.UAP_HIVI_PAGE, adUnit);
+  }
+
+  @NetworkTrafficDump(useMITM = true)
+  @Test(
+      groups = {"AdsVuapClickToPlayDesktop"}
+  )
+  public void HiViATFShouldRequestForMEGAAdUnitVAST() throws UnsupportedEncodingException {
+    final String adUnit = "/5441/wka1a.VIDEO/uap_bfab/desktop/oasis-article/_project43-life";
+    networkTrafficInterceptor.startIntercepting();
```

Comment:
Think about re-using here modified version of `checkRequestForAdUnit()`. The modification would be the third parameter which is a flag telling the method to wait or not for VAST. It will complicate the `checkRequestForAdUnit()` method a little bit but simplify `HiViATFShouldRequestForMEGAAdUnitVAST()` method which will become consistent with two other methods: `ABCDShouldRequestForMEGAAdUnitVAST()` and `HiViBTFShouldRequestForMEGAAdUnitVAST`.

Reply:
What about `checkRequestForAdUnit(page, adUnit, slotsToTrigger = [])`?